src/modules/chat/index.js

```diff
@@ -94,11 +95,11 @@ class ChatModule {
   }
 
   calculateColor(color) {
-    if (!settings.get('readableUsernameColors')) {
+    if ((settings.get(SettingIds.USERNAMES) & Usernames.READABLE) !== Usernames.READABLE) {
```

Comment:
reading/setting flags manually will only cause bugs. we should instead make utils for these operations.